StRoot/StFwdTrackMaker/FwdTrackerConfig.h

```diff
@@ -0,0 +1,224 @@
+#ifndef FWD_TRACKER_CONFIG_H
+#define FWD_TRACKER_CONFIG_H
+
+#include "TXMLEngine.h"
+#include "TString.h"
+
+#include <string>
+#include <iostream>
+#include <sstream>
+#include <vector>
+#include <map>
+#include <algorithm>
+
+// Class provides an interface for reading configuration from an XML file
+class FwdTrackerConfig {
+protected:
+
+    static const std::string valDNE; // used for nodes that DNE
+    static const std::string pathDelim; // separate node levels
+    static const std::string attrDelim; // separate attributes on nodes
+
+    bool mErrorParsing = false;
+    // read only map of the config, read with get<> functions
+    std::map<std::string, std::string> mNodes;
+    static std::stringstream sstr; // reused for string to numeric conversion
+
+    void mapFile(TXMLEngine &xml, XMLNodePointer_t node, Int_t level, std::string path = "") {
+        using namespace std;
+        // add the path delimeter above top level
+        if ( "" != path) path += FwdTrackerConfig::pathDelim;
+
+        // we skip the root node to maintain consistency with original XmlConfig
+        if ( level > 1)
+            path += xml.GetNodeName(node);
+
+        // get the node name and content if it exists
+        const string node_name = xml.GetNodeName(node);
+        const string node_content = xml.GetNodeContent(node) != nullptr ? xml.GetNodeContent(node) : FwdTrackerConfig::valDNE;
+
+        // be careful about repeated nodes
+        if ( mNodes.count( path ) == 0 ) {
+            this->mNodes[ path ] = node_content;
+        } else { // add an array index if more than one
+            path += TString::Format( "[%zu]", mNodes.count( path ) ).Data();
+            this->mNodes[ path ] = node_content;
+        }
+
+        // loop through attributes of this node
+        XMLAttrPointer_t attr = xml.GetFirstAttr(node);
+        while (attr != 0) {
+
+            // get the attribute name and value if exists
+            const string attr_name = xml.GetAttrName(attr);
+            const string attr_val = xml.GetAttrValue(attr) != nullptr ? xml.GetAttrValue(attr) : FwdTrackerConfig::valDNE;
+            
+            // save attributes with the attribute delim ":" 
+            this->mNodes[ (path + FwdTrackerConfig::attrDelim + attr_name) ] = attr_val;
+            attr = xml.GetNextAttr(attr);
+        }
+
+        // recursively get child nodes
+        XMLNodePointer_t child = xml.GetChild(node);
+        while (child != 0) {
+            mapFile(xml, child, level + 1, path);
+            child = xml.GetNext(child);
+        }
+    } // mapFile
+public:
+
+    // sanitizes a path to its canonical form 
+    static void canonize( std::string &path ) {
+        // remove whitespace
+        path.erase(std::remove_if(path.begin(), path.end(), ::isspace), path.end());
```

Comment:
`::isspace` Are you specifically calling the C function from the global namespace? Why?
```suggestion
        path.erase(std::remove_if(path.begin(), path.end(), std::isspace), path.end());
```